sessionctx/variable/sysvar.go

```diff
@@ -1674,6 +1674,11 @@ var defaultSysVars = []*SysVar{
 		s.MemoryDebugModeAlarmRatio = TidbOptInt64(val, 0)
 		return nil
 	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBDefaultStrMatchSelectivity, Value: strconv.FormatFloat(DefTiDBDefaultStrMatchSelectivity, 'f', -1, 64), Type: TypeFloat, MinValue: 0, MaxValue: 1,
+		SetSession: func(s *SessionVars, val string) error {
+			s.DefaultStrMatchSelectivity = tidbOptFloat64(val, DefTiDBDefaultStrMatchSelectivity)
```

Comment:
What about changing it to 0.1 when the set value is 0 together with a warning thrown out.

Reply:
I prefer to keep it unchanged.
If we change it to 0.1 when setting it to 0, we won't be able to know whether it's a true 0.1 or if we actually set it to 0 afterward.